dask/array/random.py

```diff
@@ -1036,37 +1037,34 @@ def _broadcast_any(ar, shape, chunks):
 """
 
 _cached_states = {}
+for _backend in ["numpy", "cupy"]:
+    with config.set({"array.backend": _backend}):
+        try:
+            _cached_states[_backend] = RandomState()
+        except ImportError:
+            pass
```

Comment:
This is probably completely safe for now. However, I'd like to avoid hard-coding the backends here if possible. Other libraries/users are technically free to expose their own backend via the entrypoint system. Therefore, we probably need to use `dask.backends.detect_entrypoints("dask.array.backends")` to detect the available backends.

With that said, I'd definitely like to avoid importing `cupy` if the user never explicitly asks to use the `cupy` backend.

Perhaps we can avoid unnecessary imports  by simply adding a lock? For example:

```python
from threading import RLock

_cached_states = {}
_cached_states_lock = RLock()

def _make_api(attr, state_constructor=None, state_class=None):
    state_constructor = state_constructor or RandomState
    state_class = state_class or RandomState

    def wrapper(*args, **kwargs):
        backend = array_creation_dispatch.backend
        key = (backend, state_constructor.__name__)

        with _cached_states_lock:
            try:
                state = _cached_states[key]
            except KeyError:
                _cached_states[key] = state = state_constructor()
            return getattr(state, attr)(*args, **kwargs)

    wrapper.__name__ = getattr(state_class, attr).__name__
    wrapper.__doc__ = getattr(state_class, attr).__doc__
    return wrapper
```

Reply:
thank you @rjzamora . Good comments especially not importing cupy (or numpy) when we are not asked for it. Used `Lock` as you suggested

I also added some comments in the code as I feel that we should not be promoting or adding to this interface to `da.random` and use `Generator` instead. Thanks again